mypyc/genops.py

```diff
@@ -1728,16 +1756,11 @@ def visit_name_expr(self, expr: NameExpr) -> Value:
             assert desc.result_type is not None
             return self.add(PrimitiveOp([], desc, expr.line))
 
-        # TODO: Behavior currently only defined for Var and FuncDef node types.
         if expr.kind == LDEF:
-            try:
-                return self.read(self.environment.lookup(expr.node), expr.line)
-            except KeyError:
-                # If there is a KeyError, then the target could not be found in the current scope.
-                # Search environment stack to see if the target was defined in an outer scope.
-                return self.read(self.get_assignment_target(expr), expr.line)
-        else:
-            return self.load_global(expr)
```

Comment:
I deleted the try-except block since `get_assignment_target` pretty much handles that now.